napari/resources/_icons.py

```diff
@@ -9,6 +9,7 @@
 
 ICON_PATH = (Path(__file__).parent / 'icons').resolve()
 ICONS = {x.stem: str(x) for x in ICON_PATH.iterdir() if x.suffix == '.svg'}
+PLUGIN_FILE_NAME = "plugin.txt"
```

Comment:
Can you elaborate a bit on this file? I'm not sure I follow what it will contain, and why it's a pure "txt".

Reply:
Name of plugi from which theme is obtained. 

To perform discovery only on this plugin, not all installed.

Reply:
Ah, I see; and how does the persistence of this file look like? Does it stick around across versions? Can/will it be deleted? What happens if one changes the name of the theme?

Reply:
In current napari the cache dir is separate per napari version.
There is no code for deleting the old cache. 

So this file will not be deleted but is visible in only one napari version. 

If the theme name is changed, then the old directory with SVG files and this file will still be present. But if someone uses an old name, then plugin discovery will fail and will be no added theme under this name.

Reply:
Ok, so there is an issue if two plugins provide a theme with the same name?

Reply:
Same as before this PR.